### PR TITLE
Move bespoke functionality out of TaskRun API

### PR DESCRIPTION
**Pull request by xchapter7x**
The TaskRun API was maintaining a `isPartOfPipeline`
function, which was only consumed by a single package (taskrunmetrics).

As the only consumer, taskrunmetrics should own this interface, and
pull the taskrun meta data it needs on its own,
rather than depending on the package to maintain a compatible API
for its bespoke use case.

closes https://github.com/tektoncd/pipeline/issues/5320

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/pipeline/v1beta1/taskrun_types_test.go [hunk 1]
```diff
@@ -217,50 +216,6 @@ func TestTaskRunHasStarted(t *testing.T) {
 	}
 }
 
-func TestTaskRunIsOfPipelinerun(t *testing.T) {
```
Comment (lbernick) [inline pkg/apis/pipeline/v1beta1/taskrun_types_test.go:220]: why not move these into metrics_test.go?

File: pkg/taskrunmetrics/metrics.go [hunk 2]
```diff
@@ -473,6 +474,17 @@ func (r *Recorder) CloudEvents(ctx context.Context, tr *v1beta1.TaskRun) error {
 	return nil
 }
 
+func isPartOfPipeline(tr *v1beta1.TaskRun) (bool, string, string) {
```
Comment (lbernick) [inline pkg/taskrunmetrics/metrics.go:477]: could you add a docstring stating that this returns the values of the pipeline label and pipelinerun label if it is part of a pipeline?

**Review by lbernick — COMMENTED** [review]
thanks John! looks good, happy to approve once tests are moved over

File: pkg/taskrunmetrics/metrics.go [hunk 3]
```diff
@@ -473,6 +474,17 @@ func (r *Recorder) CloudEvents(ctx context.Context, tr *v1beta1.TaskRun) error {
 	return nil
 }
```
Comment (JeromeJu) [inline pkg/taskrunmetrics/metrics.go:476]:
```suggestion
// IsPartOfPipeline return true if TaskRun is a part of a Pipeline.
// It also return the name of Pipeline and PipelineRun
```

**Comment by JeromeJu** [comment]
Thanks @xchapter7x for fixing this.

File: pkg/apis/pipeline/v1beta1/taskrun_types_test.go (hunk 1 above)
Comment (xchapter7x) [reply pkg/apis/pipeline/v1beta1/taskrun_types_test.go:220]: currently, the same scenarios are tested through the public API of the `taskrunmetrics` package.

I also felt that adding a test for a un-exported function would make the tests a little brittle, as I feel the implementation should be able to change below the public API without modifying the tests.

Im happy to add specific tests for this function, but would prefer to also then make it exported if we go that path.

Comment (lbernick) [reply pkg/apis/pipeline/v1beta1/taskrun_types_test.go:220]: ok that makes sense-- it does seem like there's test coverage [here](https://github.com/tektoncd/pipeline/blob/c245f3ecf9eafae7ccef1afdefbe234093baf438/pkg/taskrunmetrics/metrics_test.go#L275)

**Review by lbernick — APPROVED** [review]

**Review by jerop — COMMENTED** [review]
/lgtm

thanks @xchapter7x!

**Comment by bendory** [comment]
LGTM, and `+1` for technical hygiene!

**Pull request merged**